src/lein_monolith/plugin.clj

```diff
@@ -101,20 +101,43 @@
 
 
 (defn build-inherited-profiles
-  "Returns a map from profile keys to inherited profile maps."
+  "Returns a vector of map entries from profile keys to inherited profile maps.
+   We use a vector here instead of a map to preserve profile ordering."
   [monolith subproject]
   (reduce
     (fn [acc [key config]]
       (let [profile (some-> (choose-inheritance-source monolith config)
                             (inherited-profile subproject config)
                             (maybe-mark-leaky config))]
         (if profile
-          (assoc acc key profile)
+          (conj acc [key profile])
           acc)))
-    nil
+    []
     profile-config))
 
 
+(defn build-dependency-profiles
+  "Constructs a vector with a profile map entries containing managed dependencies from the subproject's chosen dependency set.
+   Returns nil if the subproject does not use a dependency set."
+  [monolith subproject]
+  (when-let [dependency-set (:monolith/dependency-set subproject)]
+    (let [dependencies (or (get-in monolith [:monolith :dependency-sets dependency-set])
+                           (lein/abort (format "Unknown dependency set %s used in project %s" dependency-set (:name subproject))))]
+      [[:monolith/dependency-set
+        ^:leaky {:managed-dependencies dependencies}]])))
+
+
+(defn build-profiles
+  "Constructs a vector of profile keys to inherited profile maps and the dependency set profile map.
+   We use a vector here instead of a map to preserve profile ordering."
+  [monolith subproject]
+  ;; The dependency set profile should be the first profile, so that its managed dependencies
+  ;; take precedence.
+  (concat
+    (build-dependency-profiles monolith subproject)
+    (build-inherited-profiles monolith subproject)))
```

Comment:
Docstring says this returns a vector, but it's actually returning a lazy concat. Bias for eager evaluation since we don't need laziness here (and the two inputs are already small fully-realized vectors).
```suggestion
  (vec (concat (build-dependency-profiles monolith subproject)
               (build-inherited-profiles monolith subproject))))
```